internal/services/nginx/nginx_configuration_resource.go

```diff
@@ -17,6 +17,10 @@ import (
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
 )
 
+// This resource has been deprecated and will be removed in the next major release.
```

Comment:
we should mark this resource as deprecated with instructions to use the new resource?